### PR TITLE
update to last spring boot version

### PR DESCRIPTION
**Pull request by sgrillon14**
update spring boot version 1.3.5.RELEASE to 1.5.6.RELEASE + add profile sample

### CODE REVIEW

File: spring-boot-daemon-sample/pom.xml [hunk 1]
```diff
@@ -44,6 +44,46 @@
 			<type>exe</type>
 			<scope>provided</scope>
 		</dependency>
+		
+		<!-- logback -->
+		<dependency>
+			<groupId>ch.qos.logback</groupId>
+			<artifactId>logback-core</artifactId>
+			<version>1.2.3</version>
```
Comment (snicoll) [inline spring-boot-daemon-sample/pom.xml:52]: There is no need to add those dependencies. There is a starter for logging (no version should be provided). `spring-boot-starter-web` already provides this.

File: spring-boot-daemon-sample/pom.xml [hunk 2]
```diff
@@ -44,6 +44,46 @@
...
+			<groupId>com.fasterxml.jackson.core</groupId>
+			<artifactId>jackson-core</artifactId>
+			<version>2.7.4</version>
+		</dependency>
+		<dependency>
+			<groupId>com.fasterxml.jackson.core</groupId>
+			<artifactId>jackson-annotations</artifactId>
+			<version>2.7.4</version>
+		</dependency>
+
```
Comment (snicoll) [inline spring-boot-daemon-sample/pom.xml:86]: What is this? Why do you exclude core and annotations to include them just below? Jackson is already provided by `spring-boot-starter-web` anyway and the version has to be managed by Spring Boot.

File: spring-boot-daemon-sample/src/main/java/net/nicoll/SampleApplication.java [hunk 3]
```diff
@@ -1,12 +1,9 @@
 /*
  * Copyright 2012-2015 the original author or authors.
- *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
- *
- *      http://www.apache.org/licenses/LICENSE-2.0
- *
+ * http://www.apache.org/licenses/LICENSE-2.0
```
Comment (snicoll) [inline spring-boot-daemon-sample/src/main/java/net/nicoll/SampleApplication.java]: Please don't change the header.

File: spring-boot-daemon-sample/src/main/java/net/nicoll/SampleApplication.java [hunk 4]
```diff
@@ -16,25 +13,37 @@
...
 
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.scheduling.annotation.EnableScheduling;
+import org.springframework.scheduling.annotation.Scheduled;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
 
+@EnableScheduling
```
Comment (snicoll) [inline spring-boot-daemon-sample/src/main/java/net/nicoll/SampleApplication.java:27]: I am not keen to use scheduling for the sample. That sounds like quite an agressive option to me.

File: spring-boot-daemon-sample/src/main/java/net/nicoll/SampleApplication.java [hunk 5]
```diff
@@ -16,25 +13,37 @@
...
+    }
 
-		@RequestMapping("/")
-		public String home() {
-			return "Hello World";
-		}
-	}
+    @Scheduled(fixedDelayString = "3000")
+    public void reportCurrentTime() throws InterruptedException {
+        System.out.println("---------test----------" + profile);
```
Comment (snicoll) [inline spring-boot-daemon-sample/src/main/java/net/nicoll/SampleApplication.java]: `reportCurrentTime` does not sound the right name for this method.

**Review by snicoll — CHANGES_REQUESTED** [review]
Thanks for the PR.

Can you please trim it down to only what you intend to show?

If you want to showcase how to enable a profile, there is nothing to be changed in the sample IMO. 

There are already a bunch of system property in the script so I'd rather have a section in the readme that shows how to enable a profile.

As for the rest, this looks more like "what is a profile in Spring Boot material" and it is already covered in the documentation of Spring Boot itself. 

Would you be willing to rework the PR to document this instead? Or am I missing something?

File: spring-boot-daemon-sample/src/main/java/net/nicoll/SampleApplication.java (hunk 5 above)
Comment (sgrillon14) [reply spring-boot-daemon-sample/src/main/java/net/nicoll/SampleApplication.java]: I rewrite the header (automatic change by my IDE).
I rename 'reportCurrentTime' by 'printScheduling'

File: spring-boot-daemon-sample/pom.xml (hunk 1 above)
Comment (sgrillon14) [reply spring-boot-daemon-sample/pom.xml:52]: Without this part, I have this error on runtime:

Failed to instantiate SLF4J LoggerFactory
Reported exception:
java.lang.NoClassDefFoundError: ch/qos/logback/core/joran/spi/JoranException
	at org.slf4j.LoggerFactory.bind(LoggerFactory.java:150)
	at org.slf4j.LoggerFactory.performInitialization(LoggerFactory.java:124)
	at org.slf4j.LoggerFactory.getILoggerFactory(LoggerFactory.java:412)
	at org.slf4j.LoggerFactory.getLogger(LoggerFactory.java:357)
	at org.apache.commons.logging.impl.SLF4JLogFactory.getInstance(SLF4JLogFactory.java:155)
	at org.apache.commons.logging.impl.SLF4JLogFactory.getInstance(SLF4JLogFactory.java:132)
	at org.apache.commons.logging.LogFactory.getLog(LogFactory.java:273)
	at org.springframework.boot.SpringApplication.<clinit>(SpringApplication.java:179)
	at net.nicoll.boot.daemon.SpringBootService.start(SpringBootService.java:42)
	at net.nicoll.boot.daemon.StartSpringBootService.main(StartSpringBootService.java:27)
Caused by: java.lang.ClassNotFoundException: ch.qos.logback.core.joran.spi.JoranException
	at java.net.URLClassLoader.findClass(Unknown Source)
	at java.lang.ClassLoader.loadClass(Unknown Source)
	at sun.misc.Launcher$AppClassLoader.loadClass(Unknown Source)
	at java.lang.ClassLoader.loadClass(Unknown Source)
	... 10 more
Exception in thread "main" java.lang.NoClassDefFoundError: ch/qos/logback/core/joran/spi/JoranException
	at org.slf4j.LoggerFactory.bind(LoggerFactory.java:150)
	at org.slf4j.LoggerFactory.performInitialization(LoggerFactory.java:124)
	at org.slf4j.LoggerFactory.getILoggerFactory(LoggerFactory.java:412)
	at org.slf4j.LoggerFactory.getLogger(LoggerFactory.java:357)
	at org.apache.commons.logging.impl.SLF4JLogFactory.getInstance(SLF4JLogFactory.java:155)
	at org.apache.commons.logging.impl.SLF4JLogFactory.getInstance(SLF4JLogFactory.java:132)
	at org.apache.commons.logging.LogFactory.getLog(LogFactory.java:273)
	at org.springframework.boot.SpringApplication.<clinit>(SpringApplication.java:179)
	at net.nicoll.boot.daemon.SpringBootService.start(SpringBootService.java:42)
	at net.nicoll.boot.daemon.StartSpringBootService.main(StartSpringBootService.java:27)
Caused by: java.lang.ClassNotFoundException: ch.qos.logback.core.joran.spi.JoranException
	at java.net.URLClassLoader.findClass(Unknown Source)
	at java.lang.ClassLoader.loadClass(Unknown Source)
	at sun.misc.Launcher$AppClassLoader.loadClass(Unknown Source)
	at java.lang.ClassLoader.loadClass(Unknown Source)
	... 10 more

Comment (snicoll) [reply spring-boot-daemon-sample/pom.xml:52]: Sorry but I don't see the link with the changes that you've made. I really am in favour of a documentation update only at this point.

Comment (sgrillon14) [reply spring-boot-daemon-sample/pom.xml:52]: When I Update Spring version, I have this error: java.lang.NoClassDefFoundError: ch/qos/logback/core/joran/spi/JoranException

I do not know why.

When, I return to old version, I do not need this import. I do not know why.

Comment (snicoll) [reply spring-boot-daemon-sample/pom.xml:52]: My best guess is that a jar got corrupted when it was downloaded.

**Comment by snicoll** [comment]
There is a lot going on here and I'd rather go with a documentation update for enabling a profile. There is really nothing specific here. The only thing you need to know is the property name to enable and how profiles work with Spring Boot. 

The latter is documented in Spring Boot itself and I'd rather not duplicate this here. Thanks for the PR anyway!

**Pull request closed without merging**